classes/announcement/Collector.inc.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * @file classes/announcement/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class announcement
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
+ */
+
+namespace PKP\announcement;
+
+use HookRegistry;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+
+class Collector implements CollectorInterface
+{
+    public DAO $dao;
+    public array $contextIds = [];
+    public string $searchPhrase = '';
+    public array $typeIds = [];
+    public int $count;
+    public int $offset;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Filter announcements by one or more contexts
+     */
+    public function filterByContextIds(array $contextIds): self
```

Comment:
I do prefer moving back to functions rather than a parameter array; this is less error-prone e.g. in case of a typo in defining the array contents. :+1:

Reply:
Yeah, having done this it's _soo_ much better.